### PR TITLE
new way for batching + hashing for memcached and redis

### PR DESCRIPTION
**Pull request by BoryaMogila**
new way for batching + hashing for memcached and redis

### CODE REVIEW

File: SQLCache.js [hunk 1]
```diff
@@ -1,51 +1,33 @@
...
-      Promise.all(rawQueries.map(rawQuery => knex.raw(rawQuery)))
-    );
+    this.hashing = options.hashing;
   }
 
-  getCacheKeyForQuery(query) {
+  getCacheKeyForQuery(query, ttl) {
     const queryString = query.toString();
+    if (this.hashing) {
+      return `sqlcache:${crc.crc32(queryString)}_${ttl}`;
```
Comment (cvburgess) [inline SQLCache.js:13]: does the TTL need to be in the key?

File: SQLDataSource.test.js [hunk 2]
```diff
@@ -32,32 +31,6 @@ describe("Configuration", () => {
 
     expect(initialize).toThrow(Error);
   });
-
-  test("an warning to be logged if batching is used", () => {
```
Comment (cvburgess) [inline SQLDataSource.test.js:36]: why remove these tests?

File: SQLDataSource.test.js [hunk 3]
```diff
@@ -66,6 +39,7 @@ describe("PostgreSQL", () => {
       constructor() {
         super();
         this.db = knex;
+        this.hashing = true;
```
Comment (cvburgess) [inline SQLDataSource.test.js:42]: I may be a bit confused, but if the constructor is setting this.hashing based on the config object... why is this necessary at all?

**Review by cvburgess — CHANGES_REQUESTED** [review]

**Comment by cvburgess** [comment]
Thank you for opening a PR! I left some questions and I still need to manually test this. Please reply to the comments so I can move this along.

File: SQLCache.js (hunk 1 above)
Comment (BoryaMogila) [reply SQLCache.js:13]: yes, it`s help controll dinamic cache time

File: SQLDataSource.test.js (hunk 2 above)
Comment (BoryaMogila) [reply SQLDataSource.test.js:36]: because remove raw queries and make batching like in apollo-datasource-rest. It`s show better performance for big count of queries

File: SQLDataSource.test.js (hunk 3 above)
Comment (BoryaMogila) [reply SQLDataSource.test.js:42]: if you want use in-memory cache you don`t need hashing

**Comment by cvburgess** [comment]
Can you please update this PR to work with v1.0? 

I think you may have a much easier time with less code.

**Comment by cvburgess** [comment]
This should be fixed in v1.0.1, thank you for the help, please open a PR if you can improve it or find issues.
